torch/csrc/jit/mobile/debug_info.cpp

```diff
@@ -9,8 +11,78 @@
 namespace torch {
 namespace jit {
 
+namespace {
+
+// This function construct stacktrace with module hierarchy
+// Module hierarchy will contain information about where in the
+// module hierarchy this source is. For example if conv2d op
+// exist in hierarcy A->B->C->Conv2d with type annotations of
+// A -> TopM, B->MyModule, C->SomeModule, then module hierarchy
+// will be TopM(A).MyModule(B).SomeModule(C).Conv2d(conv)
+// Source level stack information will be from model source code.
+std::pair<std::string, std::string> getStackTraceWithModuleHierarchy(
+    const DelegateDebugInfoType& sr_callstack,
+    const std::string& root_scope_string,
+    const std::string& top_module_type_name) {
```

Comment:
Why do these need to be passed in?  Why aren't they in the sr_callstack?

Reply:
Because sr_callstack contains module type information for all the submodules only. This is because during inlining we replace CallMethod nodes and for those nodes you can ask on what object type this method is defined. For top level module we cannot quite access this information besides looking at graph's primary input. Assume the first input is `self` and inspect that. While that is doable I dont think sr_callstack is the right place to record it. And I presumed most interesting thing is really within the top module.
So we can figure out some other ways to record this so that debug info captures everything and nothing has to be supplied from outside.
One thing I can think of is to create a special debug handle that captures this information.